server/src/main/java/keywhiz/service/daos/GroupDAO.java

```diff
@@ -93,22 +118,57 @@ public void deleteGroup(Group group) {
           .set(SECRETS.OWNER, (Long) null)
           .where(SECRETS.OWNER.eq(group.getId()))
           .execute();
+      DSL.using(configuration)
+          .update(GROUPS)
+          .set(GROUPS.OWNER, (Long) null)
+          .where(GROUPS.OWNER.eq(group.getId()))
+          .execute();
     });
   }
 
   public Optional<Group> getGroup(String name) {
     GroupsRecord r = dslContext.fetchOne(GROUPS, GROUPS.NAME.eq(name));
-    return Optional.ofNullable(r).map(groupMapper::map);
+    return Optional.ofNullable(toGroup(r));
   }
 
   public Optional<Group> getGroupById(long id) {
     GroupsRecord r = dslContext.fetchOne(GROUPS, GROUPS.ID.eq(id));
-    return Optional.ofNullable(r).map(groupMapper::map);
+    return Optional.ofNullable(toGroup(r));
   }
 
   public ImmutableSet<Group> getGroups() {
-    List<Group> r = dslContext.selectFrom(GROUPS).fetch().map(groupMapper);
-    return ImmutableSet.copyOf(r);
+    Result<GroupsRecord> records = dslContext.selectFrom(GROUPS).fetch();
+
+    Map<Long, Group> groupsById = records.stream()
+        .map(groupMapper::map)
+        .collect(Collectors.toMap(Group::getId, Function.identity()));
+
+    for (GroupsRecord record : records) {
+      if (record.getOwner() != null) {
+        Group group = groupsById.get(record.getId());
+        Group owner = groupsById.get(record.getOwner());
+        group.setOwner(owner.getName());
+      }
+    }
+
+    return ImmutableSet.copyOf(groupsById.values());
+  }
```

Comment:
This method returns all the groups in the database. I don't know why you'd want to do that (or why we'd allow it, especially without pagination), but as a minor optimization we're taking advantage of the fact that all groups are now in memory to populate the owner fields without having to go back to the database.